library/dokku_resource_reserve.py

```diff
@@ -0,0 +1,272 @@
+  app:
+    description:
+      - The name of the app
+    required: True
+    default: null
+    aliases: []
+  resources:
+    description:
+      - The Resource type and quantity (required when state=present)
+    required: False
+    default: null
+    aliases: []
+  process-type:
+    description:
+      - The process type selector
+    required: False
+    default: null
+    alias: []
+  clear-before:
+    description:
+      - Clear all reserves before apply
+    required: False
+    default: "False"
+    choices: [ "True", "False" ]
+    aliases: []
+  state:
+    description:
+      - The state of resources
+    required: False
+    default: present
+    choices: [ "present", "absent" ]
+    aliases: []
+author: Alexandre Pavanello e Silva
+requirements: [ ]
+
+"""
+
+EXAMPLES = """
+- name: Create a dokku app cpu and memory reserve
+  dokku_resource_reserve:
+    app: hello-world
+    resources:
+      cpu: 100
+      memory: 100
+
+- name: Create a reserve per-process type dokku app
+  dokku_resource_reserve:
+    app: hello-world
+    process-type: web
+    resources:
+      cpu: 100
+      memory: 100
+
+- name: Clear before apply new reserves
+  dokku_resource_reserve:
+    app: hello-world
+    state: present
+    clear-before: True
+    resources:
+      cpu: 100
+      memory: 100
+
+- name: Remove all resource/reserves
+  dokku_app:
+    app: hello-world
+    state: absent
+"""
+
+
+def force_list(var):
+    if isinstance(var, list):
+        return var
+    return list(var)
+
+
+def subprocess_check_output(command, split="\n"):
+    error = None
+    output = []
+    try:
+        output = subprocess.check_output(command, shell=True)
+        if isinstance(output, bytes):
+            output = output.decode("utf-8")
+        output = str(output).rstrip("\n")
+        if split is None:
+            return output, error
+
+        output = output.split(split)
+        output = force_list(filter(None, output))
+        output = [o.strip() for o in output]
+    except subprocess.CalledProcessError as e:
+        error = str(e)
+    return output, error
+
+
+def dokku_resource_clear(data):
+    error = None
+    process_type = ""
+    if data["process-type"]:
+        process_type = "--process-type {0}".format(data["process-type"])
+    command = "dokku resource:reserve-clear {0} {1}".format(process_type, data["app"])
+    try:
+        subprocess.check_call(command, shell=True)
+    except subprocess.CalledProcessError as e:
+        error = str(e)
+    return error
+
+
+def dokku_resource_reserve_report(data):
+
+    process_type = ""
+    if data["process-type"]:
+        process_type = "--process-type {0}".format(data["process-type"])
+    command = "dokku resource:reserve {0} {1}".format(process_type, data["app"])
+
+    output, error = subprocess_check_output(command)
+    if error is not None:
+        return output, error
+    output = [re.sub(r"\s+", "", line) for line in output]
+    report = {}
+
+    allowed_keys = [
```

Comment:
I see that it can be useful to have the output of the command available in parsed form - I'd just like to note that this list will need to be kept up to date with new dokku releases (if new resources are added).

Can the parsing be made agnostic of this list?